modules/angular/projects/troca-angular/src/lib/search-input/search-input.component.ts

```diff
@@ -0,0 +1,123 @@
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { Subject, Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+export interface ItemResults {
+  title: string;
+  description?: string;
+  icon?: string;
+  link?: string;
+}
+
+
+@Component({
+  selector: 'tf-search-input',
+  templateUrl: './search-input.component.html',
+  styleUrls: ['./search-input.component.scss']
+})
+
+export class SearchInputComponent {
+
+  static readonly SEARCH_MIN_CHARACTERS = 3;
+
+  @Input() value?: any;
+  @Input() placeholder?: any;
+  @Input() className?: string;
+  @Input() formControl?: FormControl;
+  @Input() formControlName?: string;
+  @Input() formGroup?: FormGroup;
+  @Input() results?: ItemResults[];
+
+  @Output() onSelect?: EventEmitter<ItemResults> = new EventEmitter<ItemResults>();
+  @Output() onChange?: EventEmitter<KeyboardEvent> = new EventEmitter<KeyboardEvent>();
+  @Output() onClear?: EventEmitter<any> = new EventEmitter<any>();
+
+  public keyboardStream = new Subject<KeyboardEvent>();
+  public keyboardObservable: Observable<KeyboardEvent> = this.keyboardStream.asObservable();
+
+  public showClearButton: boolean = <boolean>false;
+
+  public resultOnFocusIndex: number = <number>-1;
+
+  constructor() {
+
+    this.keyboardObservable
+        .pipe(
+          tap(this.nextResultFilter),
+          tap(this.previousResultFilter),
+          tap(this.goToResultFilter),
+          tap(this.updateClearInputButtonFilter),
+          tap(this.refreshResultsFilter),
+        )
+        .subscribe();
+  }
+  public getInputCharacters = (keyboardEvent: KeyboardEvent) => {
+    this.keyboardStream.next(keyboardEvent);
+  }
+
+  public nextResultFilter = (keyboardEvent) => {
+    const isLastItem = (this.resultOnFocusIndex >= this.results.length - 1);
+
+    if (keyboardEvent.key !== 'ArrowDown' || isLastItem) {
+      return;
+    }
+
+    this.resultOnFocusIndex++;
+  }
+
+  public previousResultFilter = (keyboardEvent: KeyboardEvent) => {
+
+    const isFirstItem = (this.resultOnFocusIndex < 1);
```

Comment:
Una boludez nomas:

hacer `const isFirstItem = (this.resultOnFocusIndex <= 0);`